react/src/lib/components/DeckGLMap/layers/colormap/colormapLayer.ts

```diff
@@ -153,6 +153,18 @@ export default class ColormapLayer extends BitmapLayer<
             },
             moduleParameters: mergedModuleParams,
         });
+
+        const legendProps = [];
```

Comment:
this should not be done in draw method, appropriate place to set this state would have been in onDataLoad or initializeState